LCD Interfacing/LCD.h

```diff
@@ -0,0 +1,62 @@
+#include "LPC17xx.h"
+#include "stdint.h"
+
+/*
+LCD-----ARM Pins ------- Base Board
+RS-----	P1.24 -------------P2.16
+R/W----	P1.23 -------------P3.4
+EN-----	P1.22 -------------P3.8
+D4-----	P2.4 --------------P3.3
+D5-----	P2.5 --------------P3.5
+D6-----	P2.6 --------------P3.6
+D7-----	P2.7 --------------P3.7
+*/
+
+/* Bit-shift */
+#define BS(n) (1<<(n))
+
+#define LCD_DATA_PINS LPC_GPIO2
+#define LCD_COMMAND_PINS LPC_GPIO1
+
+#define LCD_D4 4
+#define LCD_D5 5
+#define LCD_D6 6
+#define LCD_D7 7
+
+#define LCD_EN 22
+#define LCD_RW 23
+#define LCD_RS 24
+
+/* Select LCD registers */
+#define LCD_CMD		0
+#define LCD_DATA	1
+
+/* LCD Modes */
+#define LCD_MODE_1	0x30 // 8 bit and 1 line

```

Comment:
For macro's enclose all values in braces. Eg: #define LCD_MODE_1    ((UNSIGNED8)0x30)

Place comments on top of each macro and not beside it. All comments should be in /* comment */

Reply:
Macros having just numbers need not be placed in braces, but if it contains some assignment operation then it would be needed.

Reply:
Actually it's a good practice to not leave precedence related things on the compiler. It improves readability. And in this case the macro can be used that way in code that precedence may come into the picture. That's why its always advisable to use braces for all macros

Reply:
What precedence would this effect ? Because we are using this as a command to the LCD module and not for some mathematical equation. So in such case how would that effect ?